### ISSUE TITLE
open browser when run on mac

### ISSUE THREAD

**Issue by pjebs** [gohugoio/hugo#5595]
Here is an idea:
If the `GOOS` is `windows` or `darwin`, when you run `hugo server`, it should open up the browser page: `http://localhost:1313/` using package: `github.com/pkg/browser`

**Comment by pjebs** [comment]
https://github.com/gohugoio/hugo/blob/master/commands/server.go#L443

Should I submit a PR?

**Comment by bep** [comment]
No, I don't think that is a good idea, but I'll leave this issue open if others want to chime in.

**Comment by pjebs** [comment]
the idea was to address something that always annoyed me (minor irritation). I only see positives and no negatives. 

* No-one runs an actual production server on windows or mac.
* It doesn't break backward compatibility promises because the documentation never promised it.


What about a new flag that opens the browser up?

**Comment by RealOrangeOne** [comment]
I quite like this idea, but it should be an opt-in feature, and not happen by default. Assuming it's smart enough to just open a new tab in an existing browser rather than forcefully spawning a new one. 

Side note: Is there a reason this can't be done on the other supported platforms too?

**Comment by felicianotech** [comment]
I agree with the orange colored one. If this was done at all, should be opt-in. I don't like the idea of a new tab opening automatically.

I would assume then a `hugo serve` flag.

**Comment by pjebs** [comment]
Give me a flag name and I'll make a PR

**Comment by felicianotech** [comment]
Well I use GitHub's `hub` CLI a lot and their command is `hub browse` to open a repository in the browser.

So I vote for a `--browse` flag.

`hugo serve --browse` would generate the site and open it up (localhost:1313 by default) in the browser.

**Comment by pjebs** [comment]
`browse` is a bit ambiguous. It could me browse files or browse website.
I think `browser` is better.

`hugo server -v --browser`

**Comment by felicianotech** [comment]
Sounds good to me.

**Comment by pjebs** [comment]
> No, I don't think that is a good idea, but I'll leave this issue open if others want to chime in.

@bep please close the issue. don't waste anyone's time keeping it open under the pretence that it is in theory possible to flip your opinion on the matter.

**Comment by peaceiris** [comment]
@pjebs @RealOrangeOne @felicianotech 

I implemented this feature just for me.

You can try `hugo server --openBrowser` on macOS, Windows, and Linux. (I tested on only macOS)

👉 https://github.com/peaceiris/hugo/commit/3e9e60fe994e146dcd8744276a8b257154824b33

**Comment by DZuz14** [comment]
Gatsby and plenty of other generators do this. Its so simple that I have no idea why you wouldn't add it.

**Comment by pawelgrzybek** [comment]
Hi.

Thats probably not going to change your opinion @bep but personally I would find this feature useful too. I come from a JavaScript / Node.js space, and majority of the server tools implement this feature under the optional `--open` flag. Would be cool to see that as part of a hugo one day.

Thanks for building an amazing product!
